layer.yaml

```diff
@@ -1,31 +1,59 @@
-includes: ['layer:hadoop-client', 'interface:spark', 'interface:benchmark']
+includes:
+- layer:hadoop-client
+- interface:spark
+- interface:benchmark
 options:
-  hadoop-client:
+  apache-spark:
```

Comment:
I completely missed that you manually pull out the layer options in `reactive/spark.py`, so let me rephrase that: this isn't how layer options are *supposed* to work.  :)  Specifically, layer options are intended to be a method for charm layers to communicate to base layers and provide them data and otherwise modify their behavior.

There's no reason that this layer can't define its own options that it uses, but they should still be namespaced under `apache-spark` and should use the `defines` section to describe how they should be validated and defaults to give them values.

That said, since the Kafka charm layer also uses these options from `hadoop-client`, my inclination is to move the implementation for them to that layer, which unfortunately would mean removing the separation (livy vs spark) that you added, which does seem nice to have.